plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/HiveTransaction.java

```diff
@@ -63,13 +65,29 @@ public AcidTransaction getTransaction()
 
     public ValidTxnWriteIdList getValidWriteIds(HiveMetastoreClosure metastore, HiveTableHandle tableHandle)
     {
+        List<SchemaTableName> lockedTables = ImmutableList.of();
+        List<HivePartition> lockedPartitions = ImmutableList.of();
+
+        boolean partitioned = !tableHandle.getPartitionColumns().isEmpty();
+        if (partitioned) {
+            if (tableHandle.getPartitions().isPresent()) {
```

Comment:
You can fuse the IFs seince `else` branches are the same.

(Then it's IMO natural to make the lockedTables,lockedPartitions variables effectively final)